inst/@sym/subsasgn.m

```diff
@@ -513,8 +513,78 @@
 %! a(1) = [];
 %! assert( isequal( a, b));
 
+%!test
+%! % indexing with a dimension more than 2
+%! a = sym ([1, 2; 3, 4; 5 6]);
+%! indices = [1 2; 3 4; 5 6];
+%! b = [10 11; 12 13; 14 15];
+%! a (indices) = b;
+%! assert (isequal (a, sym (b)));
+
 %!error <null assignment>
 %! a = sym([1, 2; 3, 4]);
 %! a(1, 2) = [];
 
+%!test
+%! A = sym ([0 0 0]);
+%! indices = [false true false];
+%! A (indices) = 1;
+%! assert (isequal (A, sym ([0 1 0])));
+%! A (indices) = [];
+%! assert (isequal (A, sym ([0 0])));
+%! indices = [false false];
+%! A (indices) = [];
+%! assert (isequal (A, sym ([0 0])));
+
+%!shared a, b
+%! a = [1 2 3 5;4 5 6 9;7 5 3 2];
+%! b = sym(a);
+
+%!function test_bool (a, b, c)
+%!  a(c) = 0;
+%!  b(c) = 0;
+%!  assert (nnz (a == 0), nnz (b == 0))
+%!endfunction
+
+%!test
+%! c = false;
+%! test_bool (a, b, c)
+%! test_bool (a, b, c | true)
+
+%!test
+%! c = [false true];
+%! test_bool (a, b, c)
+%! test_bool (a, b, c | true)
+%! test_bool (a, b, c & false)
+
+%!test
+%! c = [false true false true; true false true false; false true false true];
+%! test_bool (a, b, c)
+%! test_bool (a, b, c | true)
+%! test_bool (a, b, c & false)
+
+%!test
+%! c = [false true false true false];
+%! test_bool (a, b, c)
+%! test_bool (a, b, c | true)
+%! test_bool (a, b, c & false)
+
+%!test
+%! c = [false; true; false; true; false];
+%! test_bool (a, b, c)
+%! test_bool (a, b, c | true)
+%! test_bool (a, b, c & false)
+
+%!test
+%! c = [false true; false true; true false];
+%! test_bool (a, b, c)
+%! test_bool (a, b, c | true)
+%! test_bool (a, b, c & false)
+
+%!test
+%! c = [2 true; true 3; true true];
```

Comment:
uu, true, when we mix logical with doubles octave transform logical values to that type.

mmm, i think we need to be careful and check how this works when we send a symbolic index matrix.